src/main/java/software/amazon/encryption/s3/materials/S3Keyring.java

```diff
@@ -63,6 +73,8 @@ public EncryptionMaterials onEncrypt(EncryptionMaterials materials) {
         }
     }
 
+    abstract protected GenerateDataKeyStrategy generateStrategy();
```

Comment:
nit/opinion: i prefer `generateDataKeyStrategy()`, even if it's not consistent with encrypt/decrypt (though you could add `DataKey` there too, i generally prefer being specific).

Reply:
I thought of making the method naming consistent with `encrypt/decrypt`. I'll rename `generateDataKeyStrategy()` along with `encrypt/decryt`.